htlcswitch/link.go

```diff
@@ -969,6 +969,9 @@ func (l *channelLink) loadAndRemove() error {
 // NOTE: This MUST be run as a goroutine.
 func (l *channelLink) htlcManager() {
 	defer func() {
+		// Send an inactive notification when the link exits.
+		l.cfg.NotifyInactiveChannel(*l.ChannelPoint())
```

Comment:
Ah, I see the issue with my original suggestion - the link can fail during reestablish and then the code in `peer/brontide.go` waits for an event that will never come. Instead of breaking the current flow that `NotifyInactiveChannel` is always called _with_ a `NotifyActiveChannel`, maybe this should be a new notification `NotifyInactiveLink`.

Reply:
cool yeah always prefer being explicit, added the suggested event now.